src/storage/s3.rs

```diff
@@ -77,6 +104,29 @@ impl S3Backend {
     pub(super) fn start_storage_transaction(&self) -> Result<S3StorageTransaction, Error> {
         Ok(S3StorageTransaction { s3: self })
     }
+
+    #[cfg(test)]
+    pub(super) fn cleanup_after_test(&self) -> Result<(), Error> {
+        if !self.temporary {
+            return Ok(());
+        }
```

Comment:
Is this check needed if there's the panic below?

Reply:
Yes, this way if we create a non-temporary S3 bucket during tests we avoid shredding data.